DataDefinitions/StarSystem.cs

```diff
@@ -130,7 +131,27 @@ public class StarSystem
 
         // Not intended to be user facing - materials available within the system
         [JsonIgnore]
-        public List<string> materialEdNames => bodies.SelectMany(b => b.materials, (b, m) => m.definition.edname).Distinct().ToList();
+        public List<string> materialEdNames
+        {
+            get
+            {
+                try
+                {
+                    List<Body> bodiesList = null;
+                    lock (bodyLock)
+                    {
+                        bodiesList = bodies;
```

Comment:
I think this just makes a new reference to the same object rather than a copy. I think we want `bodiesList = new List<string>(bodies);` to make a copy.

Reply:
I think the problem is that the List<Body> bodies collection is changing underneath us. You may be right about linking to the same object though... I'll revise to 
`                        bodiesList = new List<Body>(bodies);`